app/controllers/recipes_controller.rb

```diff
@@ -1,18 +1,46 @@
 class RecipesController < ApplicationController
   before_action :authenticate_user!, except: %i[public_index]
-  # before_action :set_recipe, only: %i[show edit update destroy]
+  before_action :set_recipe, only: %i[show edit update destroy]
 
-  def index; end
+  def index
+    @recipes = Recipe.where(user: current_user)
+  end
 
   def public_index
-    @recipes = Recipe.includes(:user, :recipe_foods).where(public: true).order('created_at DESC')
+    @recipes = Recipe.includes(recipe_foods: :food).where(public: true).order('created_at DESC')
+  end
+
+  def show
+    flash.clear
+
+    @recipe = Recipe.includes(:recipe_foods).find(params[:id])
+    @recipe_food = @recipe.recipe_foods.includes(:food)
+  end
+
+  def new
+    @recipe = Recipe.new
   end
 
-  def show; end
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
 
-  def new; end
+    if @recipe.save
+      redirect_to @recipe, notice: 'Recipe was successfully created.'
+    else
+      render :new
+    end
+  end
 
-  def edit; end
+  def update
+    @recipe.update(recipe_params)
+    redirect_to @recipe, notice: 'Recipe was successfully updated.'
+  end
```

Comment:
-[] Hello ,you have done a great job on this milestone and very impressive but the testing part not complete,the project has to test for the app to show how perfect the features of the apps is functioning fine .kindly fix it  and again I understand that your partner not show and you did all this alone but you still need to contact the student successful to help you on this because in the note you attached from the student succes they emphasis that you must complete the pair programing activities on your own ,this why I can't approve the project .the project not completed still need to write more rspec test .

Reply:
Unit tests has been added to the project for controller, views and models